src/components/search/Filter/Filter.tsx

```diff
@@ -0,0 +1,185 @@
+    }
+
+    setSelectedFilters([...selectedFilters, item])
+  }
+
+  const onCheck = ({ key, value }: IStoreSelectedFacet) => {
+    if (isMobile) {
+      onFilterChange({ key, value })
+
+      return
+    }
+
+    toggleFacet({ key, value })
+    onFilterChange({ key, value })
+  }
+
+  const Filters = () => {
+    return (
+      <div data-store-filter data-testid={testId}>
+        <UIAccordion indices={expandedIndices} onChange={onAccordionChange}>
+          {facets
+            .filter((facet) => facet.type === 'BOOLEAN')
+            .map(({ label, values, key }, index) => (
+              <UIAccordionItem key={`${label}-${index}`}>
+                <UIAccordionButton data-testid="filter-accordion-button">
+                  {label}
+                  <UIIcon
+                    component={
+                      expandedIndices.has(index) ? <div>-</div> : <div>+</div>
+                    }
+                  />
+                </UIAccordionButton>
+                <UIAccordionPanel>
+                  <UIList>
+                    {values.map((item) => {
+                      const id = `${label}-${item.label}`
+
+                      return (
+                        <li key={id}>
+                          <Checkbox
+                            id={id}
+                            checked={selectedFilters.some(
+                              (filter) => filter.value === item.value
+                            )}
+                            onChange={() => onCheck({ key, value: item.value })}
+                            data-testid="filter-accordion-panel-checkbox"
+                            data-value={item.value}
+                            data-quantity={item.quantity}
+                          />
+                          <label htmlFor={id}>
+                            {item.label} ({item.quantity})
+                          </label>
+                        </li>
+                      )
+                    })}
+                  </UIList>
+                </UIAccordionPanel>
+              </UIAccordionItem>
+            ))}
+        </UIAccordion>
+        {isMobile && (
+          <div>
+            <Button
+              onClick={() => {
+                toggleFacets(selectedFilters)
+                setExpandedIndices(new Set([]))
+                setSelectedFilters([])
+              }}
+            >
+              Clear All
+            </Button>
+            <Button
+              data-testid="apply-filters-button"
+              onClick={() => toggleFacets(selectedFilters)}
+            >
+              View Results
+            </Button>
+          </div>
+        )}
+      </div>
+    )
+  }
+
+  return isMobile ? (
+    <UIModal isOpen={isOpen} onDismiss={onDismiss}>
+      <h2>Filters</h2>
+      <Button onClick={onDismiss} data-testid="close-filters-button">
+        X
+      </Button>
+      <Filters />
+    </UIModal>
+  ) : (
+    <>
+      <h4>Filters</h4>
+      <Filters />
+    </>
+  )
```

Comment:
It seems strange to me that the component itself decides if it's going to be at a Modal or not. It looks like this should be its parent's responsibility. What do you think?